scripts/validationAndPreparation.sh

```diff
@@ -1,45 +1,64 @@
+#!/bin/bash
 
-FILES=`lftp -e "set ssl:key-file ~/.ssh/id_rsa; cls; bye" sftp://josojo:@trusted-setup.staging.gnosisdev.com`
+FILES=`lftp sftp://validationworker:@trusted-setup.staging.gnosisdev.com -e 'set sftp:connect-program "ssh -a -x -i /root/.ssh/id_rsa_worker";cls;bye'`
+
+#no contribution in folder challenges expected
+FILES=${FILES//$challenges/}
+NEWESTDATE=`cat /app/config/lastestContributionDate.txt`
 if [[ -z "${NEWESTDATE}" ]]; then
   NEWESTDATE=0
 fi
+
+TRUSTEDSETUPTURN=`cat /app/config/lastestContributionTurn.txt`
 if [[ -z "${TRUSTEDSETUPTURN}" ]]; then
   TRUSTEDSETUPTURN=1
 fi
+
 unset NEWESTFILE
+
 for f in $FILES
 do
 	echo "Processing $f"
-	DATE=` lftp -e "set ssl:key-file ~/.ssh/id_rsa; cls -l --time-style=%FT%T $f/* --sort=date | head -1; bye" sftp://josojo:@trusted-setup.staging.gnosisdev.com | awk '{print $6}' | sed 's/[^0-9]*//g'`
+	DATE=`lftp sftp://validationworker:@trusted-setup.staging.gnosisdev.com -e 'set sftp:connect-program "ssh -a -x -i /root/.ssh/id_rsa_worker"; cls -l --time-style=%FT%T '$f'/* --sort=date | head -1; bye' | awk '{print $6}' | sed 's/[^0-9]*//g'`
 	echo "DATE is $DATE"
-	if [ $DATE -ge $NEWESTDATE ]; then
-		echo "found newer date"
+	if [ $DATE -gt $NEWESTDATE ]; then
+		echo "found newer contribution"
 		NEWESTDATE=$DATE
-		NEWESTFILE=`lftp -e "set ssl:key-file ~/.ssh/id_rsa; cls -l --time-style=%FT%T $f/* --sort=date | head -1; bye" sftp://josojo:@trusted-setup.staging.gnosisdev.com | awk '{print $7}'`
-		echo "newFileName $NEWESTFILE"
+		NEWESTFILE=`lftp sftp://validationworker:@trusted-setup.staging.gnosisdev.com -e 'set sftp:connect-program "ssh -a -x -i /root/.ssh/id_rsa_worker"; cls -l --time-style=%FT%T '$f'/* --sort=date | head -1; bye' | awk '{print $7}'`
+		echo "newest contribution is $NEWESTFILE"
 	fi	
 done
 
+#If a new contribution is found, do verification and preparation for next step
 if [[ !  -z "${NEWESTFILE}" ]]; then
+	cd /app/
 	echo "starting download; this could take a while..."
-	sftp josojo@trusted-setup.staging.gnosisdev.com:${NEWESTFILE} ~/
+	sftp -i /root/.ssh/id_rsa_worker validationworker@trusted-setup.staging.gnosisdev.com:${NEWESTFILE} ~/.
 
 	echo "verifying the submission; this could take a while..."
-	cargo run --release --bin verify_transform
-
+	if [[ ! -z "${CONSTRAINED}" ]]; then
+		cargo run --release --bin verify_transform_constrained
+	else
+		cargo run --release --bin verify_transform
```

Comment:
What happens if verification fails? Do we fail the entire script and try the same solution again later? Or do we mark the solution somehow as incorrect?

Reply:
Yes, currently it would try to run it over and over again until another newer file is submitted.

I mean you have a strong point, that is huge source of resources if the participant does not do a contribution.

Probably, the easiest fix would be to store the date of the newest tested file and only check newer contributions.

Reply:
Let me know what you think about the fix

Reply:
Like it. But I'm concerned it will not persist over job restarts.

Reply:
I exported the variables so that it persists over job restarts.